src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -168,6 +169,40 @@ ov::npuw::JustInferRequest::JustInferRequest(const std::shared_ptr<ov::npuw::Com
         m_subrequests_gio.at(sub_idx).global_results[i] = out_idx;
     }
     // }}}
+
+    // Fill weights bank if applicable
+    if (!m_npuw_model->m_weights_bank_name.empty()) {
+        // FIXME: mostly copied from unpack_closure()
+        for (size_t i = 0; i < m_num_submodels; i++) {
+            LOG_INFO("Filling weights bank for Subgraph[" << i << "]...");
+            LOG_BLOCK();
+            auto& comp_model_desc = m_npuw_model->m_compiled_submodels[i];
+
+            if (!comp_model_desc.compiled_model || (comp_model_desc.replaced_by && comp_model_desc.replaced_by != i)) {
+                OPENVINO_THROW("FATAL: weights bank encountered folded submodel!");
+            }
```

Comment:
Not sure if we need to put some strict limitation here.

If the model "optimized out" (see how it is checked in code) - just skip it.

Speaking of Banks as Owners of weights, it should work even for the folded models so "replaced_by" check can be also removed.

Reply:
Done